components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -2459,7 +2459,7 @@ public static NativeArray jsFunction_getAllSubscriptions(Context cx,
                                                              Scriptable thisObj, Object[] args, Function funObj)
             throws ScriptException, APIManagementException {
 
-        if (args == null || args.length == 0 || !isStringArray(args)) {
+        if (args == null || args.length == 0) {
```

Comment:
Since you are expecting at least 3 elements in the array, you should check if the array length is < 3 for throwing an error. Otherwise an ArrayIndexOBE exception will be thrown in errornous input cases.